src/math.ts

```diff
@@ -63,26 +63,43 @@ const adjustXYWithRotation = (
   angle: number,
   deltaX: number,
   deltaY: number,
+  isResizeFromCenter: boolean,
 ) => {
   const cos = Math.cos(angle);
   const sin = Math.sin(angle);
   deltaX /= 2;
   deltaY /= 2;
   if (side === "e" || side === "ne" || side === "se") {
-    x += deltaX * (1 - cos);
-    y += deltaX * -sin;
+    if (isResizeFromCenter) {
+      x += deltaX;
+    } else {
+      x += deltaX * (1 - cos);
+      y += deltaX * -sin;
+    }
   }
   if (side === "s" || side === "sw" || side === "se") {
     x += deltaY * sin;
     y += deltaY * (1 - cos);
+    if (isResizeFromCenter) {
+      x += deltaY * -sin;
```

Comment:
Same applies here.
```js
x += deltaY * sin;
x += deltaY * -sin;
```
I guess we want if-else for all cases.

Reply:
done. Simplification for all sides! 🚀